### DISCUSSION TITLE
ERROR: API not available at: http://127.0.0.1:4005

### DISCUSSION THREAD

**Discussion by nunesgh** [searx/searx#2954] [Installation issue: Official wiki]
Hi there!

I have been trying to setup a searx instance using a subdomain. I already have apache2 installed on the server and I use it to serve two other websites, one of them being the root to the subdomain I want to use for searx.

So I have cloned the repository and changed the `SEARX_URL` variable in `utils/brand.env` to the desired subdomain (I have already tried without changing anything with no success). Then I ran `sudo -H ./utils/searx.sh install all`, which completed successfully. After that, `sudo -H utils/searx.sh inspect service` returned the following:

```
service status & log
====================

sourced .config.sh :

  PUBLIC_URL          : https://search.nunesgh.com
  SEARX_URL_PATH      : /
  SEARX_INSTANCE_NAME : searx@search.nunesgh.com 
  SEARX_INTERNAL_HTTP  : 127.0.0.1:8888

INFO:  Service account searx exists.
INFO:  ~searx: python environment is available.
INFO:  ~searx: Searx software is installed.
INFO:  uWSGI app searx.ini is enabled.
INFO:  public URL   --> https://search.nunesgh.com
INFO:  internal URL --> http://127.0.0.1:8888
INFO:  got 200 from http://127.0.0.1:8888
INFO:  got 429 from https://search.nunesgh.com
```

The `429` HTTP response is expected since no new apache2 site was created yet (this is the only difference from when I tried installing without changing the `SEARX_URL` variable).

Then I ran `sudo -H ./utils/filtron.sh install all`, which also completed successfully. But now (and in the previous installation attempt), `sudo -H utils/filtron.sh inspect service` returned the following:

```
service status & log
====================

sourced .config.sh :

  PUBLIC_URL          : https://search.nunesgh.com
  PUBLIC_HOST         : search.nunesgh.com
  FILTRON_URL_PATH    : /
  FILTRON_API         : 127.0.0.1:4005
  FILTRON_LISTEN      : 127.0.0.1:4004
  FILTRON_TARGET      : 127.0.0.1:8888

INFO:  service account filtron available.
INFO:  ~filtron: go is installed
INFO:  ~filtron: filtron app is installed
INFO:  got 404 from http://127.0.0.1:4005
ERROR: API not available at: http://127.0.0.1:4005
INFO:  got 429 from http://127.0.0.1:4004
INFO:  got 200 from http://127.0.0.1:8888
INFO:  Filtron's target is available at: http://127.0.0.1:8888
INFO:  got 429 from https://search.nunesgh.com
INFO:  public URL   --> https://search.nunesgh.com
INFO:  internal URL --> http://127.0.0.1:4004
```

This is similar to the response I received when I tried installing without changing the `SEARX_URL` variable (the only difference was a `301` HTTP error instead of a `429` from `https://search.nunesgh.com`).

But I cannot find any information about this `ERROR: API not available at: http://127.0.0.1:4005`, anywhere. What is going on here? And how can I solve this?

Thank you for your attention.

**Comment by vbuser2004** [comment]
Not sure I understand the issue - is the subdomain not working? 

Fwiw, I am also using a sub domain and have exact same results as you from "inspect service". However, when I curl the internal url it does log it correctly (as a bot), so everything appears to be working.  

Also, the domain needs to be updated in the settings.yml file, not brand.env, to take effect. It looks like you did that based on your output (i.e. search.nunesgh.com). If you can't see it I would check the apache configuration (I only know nginx) but the default is '/searx'.

**Reply by nunesgh** [reply]
Thank you for your reply, @vbuser2004 !

The documentation is currently a nightmare. Which `settings.yml` file should I update? The one inside the cloned Git folder or the one at `/etc/searx/`?

I have tried configuring the `settings.yml` file inside the cloned Git folder, but changes are not reflected in my instance, which is actually being resolved at `/searx` instead of directly at the domain or subdomain. If I change the file at `/etc/searx`, depending on what I change, the server then returns an 'Internal Server Error'.

I see no clear way of properly configuring my instance.

**Comment by vbuser2004** [comment]
The `/etc/searx/settings.yml` file is the one that matters. The only other file that you should change is the `./.config` file, updating the public url setting, but it looks like you did this based on the output.

If you can't access your site from the internet then I suspect this is an apache settings issue. All your output looks correct from the 'inspect service' script, and is the same as what I see. Can you share the apache site settings file? 

Also, here is a walk through for setting up Searx that you may find helpful:

[https://www.vultr.com/docs/install-searx-with-nginx-on-ubuntu-20-10](https://www.vultr.com/docs/install-searx-with-nginx-on-ubuntu-20-10)

PS **GO CORINTHIANS** :soccer:

**Reply by nunesgh** [reply]
I was able to access searx from the internet, but only as a directory `domain.tld/searx`, not directly from the subdomain, as configured. Also, **all** the domains hosted in my server started redirecting `domain.tld/searx` to my searx instance, instead of only the subdomain (or even the domain) that I had configured in `/etc/searx/settings.yml`.

I have deleted my searx installation since the last post, but my `/etc/apache2/sites-*/searx.conf` and `/etc/apache2/sites-*/morty.conf` files were almost like those shown [here](https://searx.github.io/searx/admin/installation-apache.html#apache-reverse-proxy), under _searx via filtron plus morty_. The difference is in a few more lines loading some modules before declaring the `Location`.

It seems to me that those `*.conf` files, automatically created by the scripts `sudo -H ./utils/searx.sh install all`, `sudo -H ./utils/filtron.sh install all`, and `sudo -H ./utils/morty.sh install all`, are not correct. They do not declare a `VirtualHost` and a `ServerName`, which I have tried to manually add but ended up crashing Apache when restarting it. I have found two other tutorials for searx on Apache that declare both `VirtualHost` and `ServerName`, but I have not tried to follow them ([here](https://skorotkiewicz.github.io/techlog/install-searx-on-apache2-and-uwsgi/) and [here](https://linuxreviews.org/Searx)).

Finally, I could not manage to declare all possible server and engines configurations in the `/etc/searx/settings.yml`. file without crashing searx.

Thank you so much for your support, @vbuser2004!

PS: Here in Minas Gerais it would be either **GO CRUZEIRO** or **GO GALO**! :D
